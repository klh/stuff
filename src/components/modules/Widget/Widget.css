/*
Nothing here - in a fully developed DX this would be picked up in the webpack flow and compiled, parsed accordingly.
The idea is here to only have styles responsible for the component itself reside with the component BEM/Atomic style

Makes things much more maintainable
*/
.widget {
    margin: 10px;
    width: 300px;
}
