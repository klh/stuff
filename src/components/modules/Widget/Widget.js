import React from 'react';
import PropTypes from 'prop-types';
import {LookupField} from '../../elements';
import './Widget.css';

/**
 * Widget as Stateless Element
 * I'd rather use REACT.MEMO here but jest is not supporting it for now.
 *
 */

//Functional Component
const Widget = (props) => (
  <div className="widget">
    <div className="panel panel-info">
      <div className="panel-heading">
        Weather in <b>{props.city}</b>
      </div>
      <ul className="list-group">
        <li className="list-group-item">
          Temperature: <b>{props.temp}Â°C</b>
        </li>
        <li className="list-group-item">
          Humidity: <b>{props.humidity}</b>
        </li>
        <li className="list-group-item">
          Wind: <b>{props.wind}</b>
        </li>
        <li className="list-group-item">
          <LookupField
            submitHandler={props.submitHandler}
            defaultValue={props.city}
          />
        </li>
      </ul>
    </div>
  </div>
);

//PropTypes
Widget.propTypes = {
  submitHandler: PropTypes.func,
  city: PropTypes.string.isRequired,
  temp: PropTypes.number.isRequired,
  humidity: PropTypes.number.isRequired,
  wind: PropTypes.string.isRequired
};

// Sane defaults
Widget.defaultProps = {
  city: '--',
  temp: 0,
  humidity: 0,
  wind: '--',
  submitHandler: () => {
    console.log('nothing bound on submit');
  }
};

//etc

export default Widget;
