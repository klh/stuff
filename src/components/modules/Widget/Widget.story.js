import React from 'react';
import {storiesOf} from '@storybook/react';
import {action} from '@storybook/addon-actions';
import withDocs from 'storybook-readme/with-docs';
import WidgetReadme from './README.md';
import Widget from './';

storiesOf('Widget', module)
    .add('Primary',
        withDocs(WidgetReadme,
            () => <Widget type="primary" onClick={action('clicked')} submitHandler={() => {
            }}>Primary</Widget>
        ));
