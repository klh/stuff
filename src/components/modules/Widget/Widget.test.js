import React from 'react';
import {mount} from "enzyme/build";

import Widget from './';


describe('<Widget />', () => {

    it('Snapshot mathes', () => {
        const WidgetRender = mount(<Widget/>);
        expect(WidgetRender).toMatchSnapshot();
    });

});

