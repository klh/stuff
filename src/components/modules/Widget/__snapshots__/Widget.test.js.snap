// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Widget /> Snapshot mathes 1`] = `
<Widget
  city="--"
  humidity={0}
  submitHandler={[Function]}
  temp={0}
  wind="--"
>
  <div
    className="widget"
  >
    <div
      className="panel panel-info"
    >
      <div
        className="panel-heading"
      >
        Weather in 
        <b>
          --
        </b>
      </div>
      <ul
        className="list-group"
      >
        <li
          className="list-group-item"
        >
          Temperature: 
          <b>
            0
            Â°C
          </b>
        </li>
        <li
          className="list-group-item"
        >
          Humidity: 
          <b>
            0
          </b>
        </li>
        <li
          className="list-group-item"
        >
          Wind: 
          <b>
            --
          </b>
        </li>
        <li
          className="list-group-item"
        >
          <LookupField
            defaultValue="--"
            submitHandler={[Function]}
          >
            <form
              action="http://localhost:3000"
              className="form-inline"
              method="get"
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  defaultValue="--"
                  name="city"
                  placeholder="City"
                  type="text"
                />
              </div>
              <input
                className="btn btn-default"
                type="submit"
              />
            </form>
          </LookupField>
        </li>
      </ul>
    </div>
  </div>
</Widget>
`;
